### ISSUE TITLE
Is StreamingSubscriptionConnection api working on NTLM auth?

### ISSUE THREAD

**Issue by yecapee** [gautamsi/ews-javascript-api#65]
this is my code -

```
var Streaming = new  ews.StreamingSubscriptionConnection(exch,1);

exch.SubscribeToStreamingNotificationsOnAllFolders([1]).then(function(responses){
    console.log('oo');
    Streaming.AddSubscription(responses);
    Streaming.OnNotificationEvent.push(function(){
        console.log('aa');
    });
    Streaming.Open();
},function(responses){
    console.log('xx');
}) 
```

but when event hapen
OnNotificationEvent not work

I use ntlmXHRApi to auth Streaming 
maybe is ntlmXHRApi  not support Streaming

have any way can make Streaming working on ntlmXHRApi?

**Comment by yecapee** [comment]
maybe the point is why ntlm auth return 401 -

Request URL:https://twmail.ami.com.tw:443/Ews/Exchange.asmx
Request Method:GET
Status Code:401 Unauthorized
Response Headersview source
content-length:0
date:Mon, 20 Jun 2016 05:57:59 GMT
server:Microsoft-IIS/7.5
set-cookie:exchangecookie=21f8b9a313b947818ef6f917ac439d73; expires=Tue, 20-Jun-2017 05:57:59 GMT; path=/; HttpOnly
www-authenticate:NTLM TlRMTVNTUAACAAAAGgAaADgAAAAFgomiwax6f+Z+oTIAAAAAAAAAAMIAwgBSAAAABgGxHQAAAA9NAEUARwBBAFQAUgBFAE4ARABTAC4AVABXAAIAGgBNAEUARwBBAFQAUgBFAE4ARABTAC4AVABXAAEADgBUAFcALQBDAEEAUwAxAAQAIgB0AHcALgBtAGUAZwBhAHQAcgBlAG4AZABzAC4AYwBvAG0AAwAyAFQAVwAtAEMAQQBTADEALgB0AHcALgBtAGUAZwBhAHQAcgBlAG4AZABzAC4AYwBvAG0ABQAiAHUAcwAuAG0AZQBnAGEAdAByAGUAbgBkAHMALgBjAG8AbQAHAAgAYJ32srjK0QEAAAAA, Negotiate
x-powered-by:ASP.NET
Request Headersview source
Authorization:NTLM TlRMTVNTUAABAAAAB6IIogAAAAAoAAAAAAAAACgAAAAFASgKAAAADw==
Connection:keep-alive
Host:twmail.ami.com.tw

**Comment by gautamsi** [comment]
I do not have access to test environment with NTLM credential. it may take a while to confirm it. 

for streaming, it uses different XHR methods which does not support replacing like you do with NTLM. 

once I have access to my test environment, I can test it and try fix the issue.

**Comment by istersin** [comment]
Is there any other workaround for this, to get streaming subscriptions working with NTLM?
Or do you know when you are going to implement this?

**Comment by gautamsi** [comment]
it is taking time for me to completely rewrite the XHR and Promise api, targeting for 1.0 beta release. 
if you want to take your time to fix it, I was planning to use 'httpntlm' with 'fetch'. there is a way to use 'ntlm' library in handshake only, you can use approach similar to `async.waterfall`  and use it with `FetchStream`. 

you can look at `HangingServiceRequestBase.ts` file to find out how is this done (using different XHR than default)

**Comment by istersin** [comment]
OK. I'll try with httpntlm. 
Btw when is the 1.0 beta planned for release?

**Comment by gautamsi** [comment]
no date, once I finish #48 it will be 0.8.0 and #94 will make it 0.9.0. will close more issues with code improvement tag + bugs, it will be released beta.

**Comment by gautamsi** [comment]
#131 is merged and #94 is closed. still awaiting 0.9 push on npm. you can still build form master to use this feature.

**Comment by gautamsi** [comment]
`ews-javascript-api@next` includes these changes. reopen if still facing issue. for NTLM v2 see #173
